src/test/java/edu/ksu/canvas/attendance/submitter/CanvasAssignmentAssistantUTest.java

```diff
@@ -0,0 +1,176 @@
+    }
+
+    @Test
+    public void createAssignmentInCanvasHappyPath() throws IOException {
+        when(canvasApiWrapperService.createAssignment(any(), any(), any())).thenReturn(assignmentOptional);
+        when(attendanceSectionService.getSectionsByCourse(COURSE_ID)).thenReturn(sectionList);
+        when(assignmentService.findBySection(any())).thenReturn(attendanceAssignment);
+        when(assignmentRepository.save(attendanceAssignment)).thenReturn(attendanceAssignment);
+
+        canvasAssignmentAssistant.createAssignmentInCanvas(COURSE_ID, attendanceAssignment, oauthToken);
+    }
+
+    @Test
+    public void createAssignmentInCanvasNullReturnError() throws IOException {
+        assignmentOptional = Optional.empty();
+
+        when(canvasApiWrapperService.createAssignment(any(), any(), any())).thenReturn(assignmentOptional);
+
+        try {
+            canvasAssignmentAssistant.createAssignmentInCanvas(COURSE_ID, attendanceAssignment, oauthToken);
+            Assert.fail("Expected IOException");
+        } catch (IOException e) {
+            Assert.assertEquals("Error while creating canvas assignment for section: " + SECTION_NAME, e.getMessage());
+        }
+    }
+
+    @Test
+    public void editAssignmentInCanvasHappyPath() throws IOException {
+        when(canvasApiWrapperService.getSingleAssignment(COURSE_ID, oauthToken, CANVAS_ASSIGNMENT_ID+"")).thenReturn(assignmentOptional);
+
+        canvasAssignmentAssistant.editAssignmentInCanvas(COURSE_ID, attendanceAssignment, oauthToken);
+        verify(canvasApiWrapperService, times(1)).editAssignment(COURSE_ID.toString(), assignment, oauthToken);
+    }
+
+    @Test
+    public void editAssignmentInCanvasCanvasAssignmentNotFound() throws IOException {
+        assignmentOptional = Optional.empty();
+
+        when(canvasApiWrapperService.getSingleAssignment(COURSE_ID, oauthToken, CANVAS_ASSIGNMENT_ID+"")).thenReturn(assignmentOptional);
+
+        try {
+            canvasAssignmentAssistant.editAssignmentInCanvas(COURSE_ID, attendanceAssignment, oauthToken);
+        } catch(IOException exception ) {
+            Assert.assertEquals("Assignment not found in Canvas", exception.getMessage());
+        }
+    }
+
+    @Test
+    public void deleteAssignmentInCanvasHappyPath() throws IOException, CanvasOutOfSyncException {
+        when(attendanceSectionService.getSectionByCanvasCourseId(COURSE_ID)).thenReturn(sectionList);
+        when(assignmentRepository.findByAttendanceSection(sectionList.get(0))).thenReturn(attendanceAssignment);
+
+        canvasAssignmentAssistant.deleteAssignmentInCanvas(COURSE_ID, oauthToken);
+        verify(canvasApiWrapperService, times(1)).deleteAssignment(COURSE_ID.toString(), CANVAS_ASSIGNMENT_ID.toString(), oauthToken);
+    }
+
+    @Test
+    public void deleteAssignmentInCanvasAssignmentNotFoundError() throws IOException, CanvasOutOfSyncException {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this method to throw at most one checked exception instead of: java.io.IOException, edu.ksu.canvas.attendance.exception.CanvasOutOfSyncException [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.ome.ksu.edu/coding_rules#rule_key=squid%3AS1160)